acceptance/tests/puppet_device_test.rb

```diff
@@ -1,74 +1,81 @@
 test_name "puppet device is able to run and configure a node"
 
+tag 'server'
+
 teardown do
   on(master, "[ -f /etc/puppetlabs/puppet/puppet.conf.bak ] && mv /etc/puppetlabs/puppet/puppet.conf.bak /etc/puppetlabs/puppet/puppet.conf")
   # revert permission changes to reset state for other tests
   on(master, "puppet agent -t --server #{master}")
 end
 
-# set sensible defaults
-on(master, "cp /etc/puppetlabs/puppet/puppet.conf /etc/puppetlabs/puppet/puppet.conf.bak")
-on(master, "puppet config set server $(hostname --fqdn)")
-
-codedir = "#{master.puppet['codedir']}/modules"
-confdir = master.puppet['confdir']
-common_options="--config /tmp/puppet.conf --debug --trace -w0"
-
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
-
-apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
-# create and configure test user to trigger PUP-9642
-user {
-  'test':
-    ensure => 'present',
-}
-
-# testfile for --apply
-file {
-  '/tmp/spinner.pp':
-    content => "spinner { '999': ensure => 'absent' }"
-}
-
-# install and configure the Resource API
-service { 'puppetserver': }
-include 'resource_api::server', 'resource_api::agent'
+with_puppet_running_on(master, {}) do
+  # set sensible defaults
+  on(master, "cp /etc/puppetlabs/puppet/puppet.conf /etc/puppetlabs/puppet/puppet.conf.bak")
+  on(master, "puppet config set server $(hostname --fqdn)")
```

Comment:
I think this should happen before we start the server above.

Reply:
This seems like it might be unnecessary altogether, but I can leave it to be safe.

Reply:
If still needed. The `with_puppet_running_on` should take care of the backing up, and if necessary can also write this value.

Reply:
Did you find out if this is unnecessary?

Reply:
I think in our pipelines it's unnecessary, but with the puppet pre-suites, it's needed, I tried it. They don't set `server`.

Reply:
That is true sadly, which is why they run with `--server #{master.hostname}"`.

Reply:
I wish the tests were set up in a way that `--server` was not needed at all.

Reply:
Currently the `puppetserver ca` CLI doesn't accept `server` as a flag. I don't remember if there's a reason why it _can't_, but it's not written that way right now. So you have to do it in settings.

Reply:
I meant that even more generally: why is the puppetserver of the testsuite not already configured as `server=` in the default puppet.conf, and/or why is that puppet server not reachable as `puppet` in DNS? A lot of special casing and configuring all over the place would go away, and only the tests of the `server=` config handling and fail-over would need to concern them with it.

Reply:
Yeah when we first released our new CLI tool that relied on it being configured, we were really surprised to find that it generally isn't. In our test suites, it is part of the default.